server/service/src/localisations.rs

```diff
@@ -49,39 +75,94 @@ impl Localisations {
 
     // Get a translation for a given key and language
     // next need to add fallback and namespace to get Translation function
-    pub fn get_translation(&self, key: &str, language: &str, namespace: &str, fallback: &str ) -> String {
-        self.translations
-            .get(language)
-            .and_then(|map| map.get(namespace))
-            .and_then(|map| map.get(key))
-            .cloned()
-            .unwrap_or_else(|| fallback.to_string())
+    pub fn get_translation(&self, args: &HashMap<String, serde_json::Value> ) -> Result<String, TranslationError> {
+        let key = args.get("key").and_then(serde_json::Value::as_str).unwrap_or("").to_string();
+        let lang = args.get("lang").and_then(serde_json::Value::as_str).unwrap_or("en").to_string();
+        // use common.json namespace if nominated namespace can't be found
+        let namespace = args.get("namespace").and_then(serde_json::Value::as_str).unwrap_or("common.json").to_string();
+        let fallback = args.get("fallback").and_then(serde_json::Value::as_str);
+        match self.translations
+        .get(&lang)
+        .and_then(|map| map.get(&namespace))
+        .and_then(|map| map.get(&key))
+        .cloned() {
+            Some(translation) => Ok(translation),
+            None => {
+                // Fall back to translation from common.json if key doesn't exist in nominated namespace.
+                let common_translation = self.translations
+                    .get(&lang)
+                    .and_then(|map| map.get("common.json"))
+                    .and_then(|map| map.get(&key))
+                    .clone();
+                if let Some(common_translation) = common_translation {
+                    Ok(common_translation.to_string())
+                 } else if let Some(fallback) = fallback {
+                    Ok(fallback.to_string())
+                 } else {
+                    Err(TranslationError)   
+                 }
+            }
+        }
```

Comment:
I've updated the get_translation to a result/error structure and throw tera error if it fails to render + no fallback